Annotato/Annotato/Persistence/RootPersistenceManager.swift

```diff
@@ -0,0 +1,46 @@
+import Foundation
+import Combine
+import AnnotatoSharedLibrary
+
+class RootPersistenceManager: ObservableObject {
+    private var cancellables: Set<AnyCancellable> = []
+
+    @Published private(set) var crudDocumentMessage: Data?
+    @Published private(set) var crudAnnotationMessage: Data?
+
+    init() {
+        setUpSubscribers()
+    }
+
+    private func setUpSubscribers() {
+        WebSocketManager.shared.$message.sink { [weak self] message in
+            guard let message = message else {
+                return
+            }
+
+            self?.handleIncomingMessage(message: message)
+        }.store(in: &cancellables)
+    }
+
+    private func handleIncomingMessage(message: Data) {
+        do {
+            let decodedMessage = try JSONCustomDecoder().decode(AnnotatoMessage.self, from: message)
```

Comment:
Can we extract this out to a function that returns an optional `decodedMessage` and perform the `try-catch` in that function? That helps to remove this level of nesting. Same for the other PersistenceManagers.